bwapi/Library/BWAPILIBUnitTest/UnitTypeTest.cpp

```diff
@@ -14,866 +17,1052 @@
+}
+
+TEST_P(UnitTypeFixture, maxAirHits) {
+	UnitType u = GetParam();
+	if (u == UnitTypes::Terran_Goliath) {
+		ASSERT_EQ(u.maxAirHits(), 1);
+	}
+	else if (u == UnitTypes::Hero_Alan_Schezar) {
+		ASSERT_EQ(u.maxAirHits(), 1);
+	}
+	else {
+		ASSERT_EQ(u.maxAirHits(), BW::units.maxAirHits[u.getID()]);
+	}
+}
+
+TEST_P(UnitTypeFixture, maxGroundHits) {
+	UnitType u = GetParam();
+	if (u == UnitTypes::Terran_Goliath) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else if (u == UnitTypes::Hero_Alan_Schezar) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else if (u == UnitTypes::Terran_Siege_Tank_Tank_Mode) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else if (u == UnitTypes::Terran_Siege_Tank_Siege_Mode) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else if (u == UnitTypes::Hero_Edmund_Duke_Tank_Mode) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else if (u == UnitTypes::Hero_Edmund_Duke_Siege_Mode) {
+		ASSERT_EQ(u.maxGroundHits(), 1);
+	}
+	else {
+		ASSERT_EQ(u.maxGroundHits(), BW::units.maxGroundHits[u.getID()]);
+	}
+}
+
+TEST_P(UnitTypeFixture, mineralPrice) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.mineralPrice(), BW::units.oreCost[u.getID()]);
+}
+
+TEST_P(UnitTypeFixture, seekRange) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.seekRange(), BW::units.targetAcquisitionRange[u.getID()]*32);
+}
+
+TEST_P(UnitTypeFixture, sightRange) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.sightRange(), BW::units.sightRange[u.getID()]*32);
+}
+
+TEST_P(UnitTypeFixture, size) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.size().getID(), BW::units.sizeType[u.getID()]);
+}
+
+TEST_P(UnitTypeFixture, spaceProvided) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.spaceProvided(), BW::units.spaceProvided[u.getID()]);
+}
+
+TEST_P(UnitTypeFixture, spaceRequired) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.spaceRequired(), BW::units.spaceRequired[u.getID()]);
+}
+
+TEST_P(UnitTypeFixture, supplyProvided) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.supplyProvided(), BW::units.supplyProvided[u.getID()]);
+}
+
+TEST_P(UnitTypeFixture, supplyRequired) {
+	UnitType u = GetParam();
+	ASSERT_EQ(u.supplyRequired(), BW::units.supplyRequired[u.getID()]);
+}
+
+
 INSTANTIATE_TEST_SUITE_P(
-  UnitTypes,
-  UnitTypeFixture,
-  ::testing::ValuesIn(UnitTypeFixture::allInternalUnitTypes),
-  get_unitType_string
+	UnitTypes,
+	UnitTypeFixture,
+	::testing::ValuesIn(UnitTypeFixture::allInternalUnitTypes),
+	get_unitType_string
 );
 
 TEST_P(NoneUnknownUnitTypeFixture, data) {
-  ASSERT_EQ(GetParam().maxHitPoints(), 0);
-  ASSERT_EQ(GetParam().maxShields(), 0);
+	ASSERT_EQ(GetParam().maxHitPoints(), 0);
+	ASSERT_EQ(GetParam().maxShields(), 0);
+	if (GetParam() == UnitTypes::Unknown) {
+		ASSERT_EQ(GetParam().airWeapon(), 131);
+	}
+	else if (GetParam() == UnitTypes::None) {
+		ASSERT_EQ(GetParam().airWeapon(), 130);
+	}
+	else if (GetParam() == UnitTypes::AllUnits) {
+		ASSERT_EQ(GetParam().airWeapon(), 130);
+	}
+	else {
+		ASSERT_EQ(GetParam().airWeapon(), 130);
+	}
```

Comment:
Use `NoneUnitTypeFixture` that doesn't need if statements and just add a separate singular `TEST` for unknown.

You may need to add the corresponding fixture or `INSTANTIATE_TEST_SUITE_P`.